### PR TITLE
JCLOUDS-1598: Support Metric Alert Operation

### PR DESCRIPTION
**Pull request by SATYANAN-ANAND**
Please find this PR for Metric Alert operations:

### CODE REVIEW

**Review by gaul — APPROVED** [review]
This seems fine -- @nacx what do you think?

**Review by nacx — CHANGES_REQUESTED** [review]
Thanks, @SATYANAN-ANAND !!

We need the Mock and Live tests to be implemented for the API before we can merge the PR. Could you add those? It would be great if you could also share the output of the live tests one they're added.

Thanks!

File: providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/MetricAlert.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	public abstract String name();
+
+	/**
+	 * The location of the location
+	 */
+	public abstract String location();
+
+	/**
+	 * The type of the type
+	 */
```
[the remark this answers is not in the document]
Comment (SATYANAN-ANAND) [reply providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/MetricAlert.java]: Hi,
I have updated the javadoc and already added the MetricAlertApiMockTest.java and MetricAlertApiLiveTest.js in this PR.

[the remark this answers is not in the document]
Comment (nacx) [reply providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/MetricAlert.java]: Looks like the mock/live tests are not yet pushed?

[the remark this answers is not in the document]
Comment (SATYANAN-ANAND) [reply providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/MetricAlert.java]: Sorry! My bad. 
I added those files.

**Comment by SATYANAN-ANAND** [comment]
> Thanks, @SATYANAN-ANAND !!
> 
> We need the Mock and Live tests to be implemented for the API before we can merge the PR. Could you add those? It would be great if you could also share the output of the live tests one they're added.
> 
> Thanks!

Please find the test reports.

[MetricAlertApiLiveTest.pdf](https://github.com/apache/jclouds/files/8139363/MetricAlertApiLiveTest.pdf)
[MetricAlertApiMockTest.pdf](https://github.com/apache/jclouds/files/8139364/MetricAlertApiMockTest.pdf)

**Review by nacx — APPROVED** [review]
Thanks @SATYANAN-ANAND!! LGTM

Have you run the Live tests? Could you paste the results of running these tests in a comment?

**Pull request merged**